ProjectBank/Infrastructure.Tests/ProjectRepositoryTests.cs

```diff
@@ -4,18 +4,165 @@ public class ProjectRepositoryTests : ContextSetup, IDisposable
+        Assert.Null(await _projectRepository.ReadProjectByIdAsync(100));
+    }
+
+    [Fact]
+    public async Task ReadProjectsByTag_given_TagId_returns_list_of_ProjectDTO()
+    {
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Algo", UserId = 1, Tags = new List<string>() { "Economy" } });
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "DMAT", UserId = 1, Tags = new List<string>() { "Economy" } });
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Disys", UserId = 1, Tags = new List<string>() { "Economy" } });
+
+        var projects = await _projectRepository.ReadProjectsByTagIdAsync(1);
+
+        Assert.Collection(projects,
+        project => Assert.Equal(new ProjectDTO(1, "Algo", Status.Active, 1), project),
+        project => Assert.Equal(new ProjectDTO(2, "DMAT", Status.Active, 1), project),
+        project => Assert.Equal(new ProjectDTO(3, "Disys", Status.Active, 1), project)
+        );
+    }
+
+    [Fact]
+    public async Task ReadProjectsByUserId_given_UserId_returns_list_of_ProjectDTO()
+    {
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Algo", UserId = 1 });
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "DMAT", UserId = 2 });
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Disys", UserId = 2 });
+
+        var projects = await _projectRepository.ReadProjectsByUserIdAsync(2);
+
+        Assert.Collection(projects,
+        project => Assert.Equal(new ProjectDTO(2, "DMAT", Status.Active, 2), project),
+        project => Assert.Equal(new ProjectDTO(3, "Disys", Status.Active, 2), project)
+        );
+    }
+
+    [Fact]
+    public async Task UpdateProjectStatusById_given_ProjectId_changes_status()
+    {
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Algo", UserId = 1 });
+
+        await _projectRepository.UpdateProjectStatusByIdAsync(1);
+
+        var expected = Status.Closed;
+        var actual = _context.Projects.Find(1).Status;
+
+        Assert.Equal(expected, actual);
+
+    }
+
+    [Fact]
+    public async Task AddUserToProject_given_ProjectId_and_UserId_adds_user()
+    {
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Algo", UserId = 1 });
+        await _userRepository.CreateUserAsync(new CreateUserDTO
+        {
+            Name = "Alice",
+            Email = "email@email.com",
+            Password = "Password123",
+            Role = Role.Student
+        });
+        await _projectRepository.AddUserToProjectAsync(1, 1);
+        await _projectRepository.AddUserToProjectAsync(2, 1);
+
+        var project = await _context.Projects.FindAsync(1);
 
-        var actual = _context.Projects.Find(1).Title;
+        var actual = project.Participants.ElementAt(0);
+        var expected = await _context.Users.FindAsync(1);
```

Comment:
Why is the expected something we await?

Reply:
I think we should just run as many things async as possible hmm